### PR TITLE
#1277 only proj reimbursement

### LINKED ISSUE #1277: Finance Fixes - Only Allow Reimbursement Products to be made on projects Frontend
### Description

users want their finances tracked at a project level, not a work package level, only give the options to add wbs elements that are projects in the autocomplete options. Also Change the placeholder from Select a WBS element to Select a Project

### Acceptance Criteria

Only shows projects, WBS Element is no longer mentioned anywhere on the reimbursement request form

### Proposed Solution

Edit the autocomplete options in the reimbursement request form + Change the placeholder text for the Autocomplete + Change the WBS Element header on the product table to just be project
epic #1274 
### Mocks

_No response_

### PR DESCRIPTION
**Pull request by ScarlettBlaize**
## Changes

Changed placeholder to "Select a Project" and a form label to "Project" in the reimbursement product table file

Removed the allWbsElements constant. Created a new constant, projectDropdown, to only show projects and not work packages in the dropdown menu

## Screenshots

Screenshot 2023-09-21 004726

Screenshot 2023-09-21 004710

image

Closes #1277

### CODE REVIEW

**Comment by ninest** [comment]
Great work! Can you run `yarn lint` and make sure to make any changes required? The linting check is failing

File: src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementRequestForm.tsx
Comment (ninest) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementRequestForm.tsx:1]: Remove unsused imports. Seems like 

```
import { ClubAccount, ReimbursementProductCreateArgs, ReimbursementReceiptUploadArgs, WbsNumber } from 'shared';
```

can be changed to

```
import { ClubAccount, ReimbursementProductCreateArgs, ReimbursementReceiptUploadArgs } from 'shared';
```

File: src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementRequestForm.tsx [hunk 1]
```diff
@@ -157,10 +156,12 @@ const ReimbursementRequestForm: React.FC<ReimbursementRequestFormProps> = ({
     }
   };
 
-  const allWbsElements: {
-    wbsNum: WbsNumber;
-    wbsName: string;
-  }[] = getAllWbsElements(allProjects);
+  const projectDropdown = allProjects.map((proj) => {
```
Comment (ninest) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementRequestForm.tsx:159]: Can you rename this to something like `allProjectWbsElements` to keep it consistent with what was there previously?

**Review by ninest — CHANGES_REQUESTED** [review]
Everything looks good! Just a few small changes

**Review by ninest — APPROVED** [review]

**Review by ninest — APPROVED** [review]

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**